cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java

```diff
@@ -219,24 +219,25 @@ private static CpuCapacity getCpuBasedOnRequirements(ResourceRequirements resour
         Integer limit = getResourceRequirement(resourceRequirements, ResourceRequirementType.LIMIT);
 
         if (limit != null) {
-            if (request == null || limit.intValue() == request.intValue()) {
+            if (request == null) {
                 return new CpuCapacity(CpuCapacity.milliCpuToCpu(limit));
+            } else {
+                return new CpuCapacity(CpuCapacity.milliCpuToCpu(request));
             }
+        } else if (request != null) {
+            return new CpuCapacity(CpuCapacity.milliCpuToCpu(request));
         }
 
         return null;
     }
 
     private CpuCapacity processCpu(io.strimzi.api.kafka.model.balancing.BrokerCapacity bc, BrokerCapacityOverride override, CpuCapacity cpuBasedOnRequirements) {
-        if (cpuBasedOnRequirements != null) {
-            if ((override != null && override.getCpu() != null) || (bc != null && bc.getCpu() != null)) {
-                LOGGER.warnCr(reconciliation, "Ignoring CPU capacity override settings since they are automatically set to resource limits");
-            }
-            return cpuBasedOnRequirements;
-        } else if (override != null && override.getCpu() != null) {
+        if (override != null && override.getCpu() != null) {
             return new CpuCapacity(override.getCpu());
         } else if (bc != null && bc.getCpu() != null) {
             return new CpuCapacity(bc.getCpu());
+        } else if (cpuBasedOnRequirements != null) {
+            return cpuBasedOnRequirements;
         } else {
             return new CpuCapacity(BrokerCapacity.DEFAULT_CPU_CORE_CAPACITY);
         }
```

Comment:
Should the logic here be changed? You now do this:
1) Process the CPU based on the resources
2) If overrides are set, you just throw it away and use the overrides

What would make more sense is this:
1) Check if overrides are set and use them if they are
2) If they are not set, process the CPU based on the resource requirements

Reply:
Logic changed